src/System.Runtime/tests/System/TypeTests.cs

```diff
@@ -8,6 +8,9 @@
 using System.Reflection;
 using Xunit;
 
+// These classes conflict with ones from System.Reflection/tests/ModuleTests.cs
+// within single test assembly on MONO.
+#if !MONO
```

Comment:
I'd prefer renaming the classes so they're unique. The definitions in ModuleTests.cs might change and then they could become incompatible.

Reply:
Then it makes sense to rename classes in ModuleTests.cs because it contains less occurrences to rename.

Reply:
Sure. Please also upstream this change :)

Reply:
yeah, working on it

Reply:
Done

Reply:
@akoeplinger Could you take another look?